cl/corpus_importer/tasks.py

```diff
@@ -2005,12 +2005,21 @@ def download_pacer_pdf_by_rd(
     s = ProxyPacerSession(
         cookies=session_data.cookies, proxy=session_data.proxy_address
     )
-    report = FreeOpinionReport(pacer_court_id, s)
-
-    r, r_msg = report.download_pdf(
-        pacer_case_id, pacer_doc_id, magic_number, de_seq_num=de_seq_num
-    )
-
+    if is_appellate_court(pacer_court_id):
+        report = AppellateDocketReport(pacer_court_id, s)
+        pacer_doc_id = (
+            pacer_doc_id
+            if not rd.attachment_number
+            else int(f"{str(pacer_doc_id)[:3]}1{str(pacer_doc_id)[4:]}")
```

Comment:
I ran a test, and it failed on this line. Here, the `pacer_doc_id` is being converted from a `str` to an `int`, which removes the leading zeros and results in an incorrect PACER query.

Additionally, the `pacer_doc_id` type in the parameters should be `str` instead of `int`.

Reply:
Thanks @albertisfu! I've updated the function signature and fixed the code to correctly handle `pacer_doc_id`. 

I also Included a test to verify the `download_pacer_pdf_by_rd` function updates the **doc_id** for appellate attachment.